### ISSUE TITLE
Validation step in a parallel stage in a scripted build returns a wrong result

### ISSUE THREAD

**Issue by dcoraboeuf** [jenkinsci/ontrack-plugin#47]
Given such a pipeline:

```groovy
node ("docker") {
    stage("Build") {
        sleep time: 1, unit: 'SECONDS'
        ontrackBuild project: 'test', branch: 'master', build: "$BUILD_NUMBER"
    }
    stage("Tests") {
        parallel([
            "Test 1": {
                stage("Test 1") {
                    node("docker") {
                        try {
                            sleep time: 5, unit: 'SECONDS'
                            sh 'exit 1'
                        } finally {
                            ontrackValidate project: 'test', branch: 'master', build: "$BUILD_NUMBER", validationStamp: 'VS1', buildResult: currentBuild.currentResult
                        }
                    }
                }
            },
            "Test 2": {
                stage("Test 2") {
                    node("docker") {
                        sleep time: 15, unit: 'SECONDS'
                        ontrackValidate project: 'test', branch: 'master', build: "$BUILD_NUMBER", validationStamp: 'VS2', buildResult: currentBuild.currentResult
                    }
                }
            }
        ])
    }
}
```

The "VS1" validation stamp for stage "Test 1" will report "Passed" when "Failed" is expected.

The current build result cannot be used, only the current stage's status can be used.

**Comment by dcoraboeuf** [comment]
Correction of status is now OK:

![image](https://user-images.githubusercontent.com/1206964/43038505-3c6e26e8-8d1a-11e8-914b-662bd0fd3848.png)

However, the refactoring of the stage detection has lost us the duration.

**Comment by dcoraboeuf** [comment]
Fixed in 2.33.3
